### PR TITLE
feat: remove country for free events

### LINKED ISSUE #375: dont show country edit text for free orders
working on it 👍

### PR DESCRIPTION
**Pull request by simarsingh24**
fixes #375

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt [hunk 1]
```diff
@@ -279,7 +285,7 @@ class AttendeeFragment : Fragment() {
                                 email = email.text.toString(),
                                 ticket = TicketId(it.first.toLong()),
                                 event = eventId)
-                        val country = country.text.toString()
+                        val country = if (country.text.toString().length > 0) country.text.toString() else null
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: Text.isEmpty

File: app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt [hunk 2]
```diff
@@ -248,6 +248,12 @@ class AttendeeFragment : Fragment() {
                 rootView.qty.text = " — $it items"
             })
 
+            attendeeFragmentViewModel.totalAmount.observe(this, Observer {
+                if (it != null && it <= 0) {
+                    rootView.country.visibility = View.GONE
+                }
+            })
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: ViewModel is there for logic, I don't get why every now and then logic is leaked to view

Imagine Activity and Fragments only contain XML and not Java, and solve problems using that idea

Comment (simarsingh24) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: resolved

File: app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt [hunk 3]
```diff
@@ -249,7 +249,10 @@ class AttendeeFragment : Fragment() {
 
             attendeeFragmentViewModel.countryVisibility.observe(this, Observer {
                 if (it != null) {
-                    rootView.countryArea.visibility = it
+                    if (it)
+                        rootView.countryArea.visibility = View.VISIBLE
+                    else
+                        rootView.countryArea.visibility = View.GONE
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: `rootView.countryArea.visibility = if (it) View.VISIBLE else View.GONE`

**Pull request merged**